### PR TITLE
README/tests: add BIP173/BIP16 SegWit address examples

### PR DESCRIPTION
**Pull request by dcousens**
@junderw please review.

And any others,  please review.

### CODE REVIEW

**Comment by dcousens** [comment]
Will release `3.2.0` after this is reviewed and merged.

**Comment by dcousens** [comment]
I might add https://github.com/bitcoinjs/bitcoinjs-lib/issues/853 to this,  to ensure these outputs are capable of being spent :+1:

File: test/integration/addresses.js [hunk 1]
```diff
@@ -49,6 +47,56 @@ describe('bitcoinjs-lib (addresses)', function () {
...
+    assert.strictEqual(address, 'bc1qt97wqg464zrhnx23upykca5annqvwkwujjglky')
+  })
+
+  it('can generate a SegWit 2-of-2 multisig address (via P2SH)', function () {
+    var pubKeys = [
+      '026477115981fe981a6918a6297d9803c4dc04f328f22041bedff886bbc2962e01',
+      '02c96db2302d19b43d4c69368babace7854cc84eb9e061cde51cfa77ca4a22b8b9'
+    ].map(function (hex) { return Buffer.from(hex, 'hex') })
+
+    var redeemScript = bitcoin.script.multisig.output.encode(2, pubKeys) // 2 of 2
```
Comment (junderw) [inline test/integration/addresses.js]: This should be witnessScript, not redeemScript

File: test/integration/addresses.js [hunk 2]
```diff
@@ -49,6 +47,56 @@ describe('bitcoinjs-lib (addresses)', function () {
...
+    var redeemScript2 = bitcoin.script.witnessScriptHash.output.encode(bitcoin.crypto.sha256(redeemScript))
```
Comment (junderw) [inline test/integration/addresses.js]: redeemScript

File: test/integration/addresses.js [hunk 3]
```diff
@@ -49,6 +47,56 @@ describe('bitcoinjs-lib (addresses)', function () {
...
+
+  it('can generate a SegWit 3-of-4 multisig address', function () {
+    var pubKeys = [
+      '026477115981fe981a6918a6297d9803c4dc04f328f22041bedff886bbc2962e01',
+      '02c96db2302d19b43d4c69368babace7854cc84eb9e061cde51cfa77ca4a22b8b9',
+      '023e4740d0ba639e28963f3476157b7cf2fb7c6fdf4254f97099cf8670b505ea59',
+      '03c6103b3b83e4a24a0e33a4df246ef11772f9992663db0c35759a5e2ebf68d8e9'
+    ].map(function (hex) { return Buffer.from(hex, 'hex') })
+
+    var redeemScript = bitcoin.script.multisig.output.encode(3, pubKeys) // 3 of 4
```
Comment (junderw) [inline test/integration/addresses.js]: witnessScript

**Comment by junderw** [comment]
Some readability changes... People wrapping their heads around the difference between redeemScript and witnessScript and witnessPubKeyHash

**Comment by dcousens** [comment]
> Some readability changes... People wrapping their heads around the difference between redeemScript and witnessScript and witnessPubKeyHash

@junderw interestingly,  this is something @afk11 and I discussed in depth in https://github.com/bitcoinjs/bitcoinjs-lib/issues/695.

I argued that in the long term,  we _shouldn't_ be distinguishing between them.
In the end,  users should only provide the required redeem scripts/data any-how.

**Comment by dcousens** [comment]
E.g,  thinking of a MAST friendly `TransactionBuilder`,  the API might be:

* addInput (txHash, vout, sequence)
* sign (vin, keyPair[, options])
  * options: { txOut, redeemScripts, hashType }
    * txOut: { script, value }

**Comment by junderw** [comment]
redeemScripts?

How would we tell the difference between SH (HASH160) and WSH (SHA256)???

redeemScripts should refer to P2SH only (hash160) and if we decide to create an all-inclusive term it should incorporate the hash algorithm.

redeemScriptHash160
redeemScriptSha256

etc. etc.

"redeemScripts" is mixing P2SH in with other things which will confuse developers new to the concept.

my two cents.

**Comment by dcousens** [comment]
@junderw to me, a `redeemScript` is a redeem "script" that I use to redeem/solve an output.
It is an out-of-band script that is not exposed until "redeemed".

But I suppose it is helpful to "encode" more context as to what layer it is via the "redeemScript -> P2SH", "witnessScript -> P2WSH", "mastScript -> ...etc".
At least until we have a better idea.

**Pull request merged**